aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterQueryTest.java

```diff
@@ -225,6 +243,299 @@ public void querySavedDataWithMultipleForeignKeysOfSameType() throws DataStoreEx
+                .name("Sample Blog")
+                .id(blogId)
+                .owner(blogOwner)
+                .build();
+
+        final Post2 post = Post2.builder()
+                .title("Placeholder title")
+                .status(PostStatus.ACTIVE)
+                .rating(5)
+                .id(postId)
+                .blog(blog)
+                .blogOwner(blogOwner)
+                .build();
+
+        adapter.save(blogOwner);
+        adapter.save(blog);
+        adapter.save(post);
+
+        final List<Post2> posts = adapter.query(Post2.class,
+                Where.matches(Post2.BLOG_OWNER.eq(blogOwnerId).and(Post2.RATING.gt(4).
+                        and(Post2.STATUS.eq(PostStatus.ACTIVE)))));
+        assertTrue(posts.contains(post));
+    }
+
+    /**
+     * Test that querying the saved item with multiple paths for foreign keys to the same
+     * model along with Multiple conditions and Custom Primary Key correctly forms the where clause
+     * and returns expected data.
+     * @throws DataStoreException On unexpected failure manipulating items in/out of DataStore.
+     */
+    @Test
+    public void querySavedDataWithMultipleJoinPathsWithMultipleConditionsAndCPK() throws DataStoreException {
+        setupForBlogCPKModel();
+        final String blogOwnerId = UUID.randomUUID().toString();
+        final String blogId = UUID.randomUUID().toString();
+        final String postId = UUID.randomUUID().toString();
+
+        final BlogOwnerCPK blogOwner = BlogOwnerCPK.builder()
+                .id(blogOwnerId)
+                .name("Sample BlogOwner")
+                .build();
+
+        final BlogCPK blog = BlogCPK.builder()
+                .id(blogId)
+                .name("Sample Blog")
+                .owner(blogOwner)
+                .build();
+
+        final PostCPK post = PostCPK.builder()
+                .id(postId)
+                .title("Placeholder title")
+                .rating(5)
+                .blog(blog)
+                .blogOwner(blogOwner)
+                .build();
+
+        adapter.save(blogOwner);
+        adapter.save(blog);
+        adapter.save(post);
+        String expectedBlogOwnerCPK = "\"" + blogOwnerId + "\"#\"Sample BlogOwner\"";
+        final List<PostCPK> posts = adapter.query(PostCPK.class,
+                Where.matches(PostCPK.BLOG_OWNER.eq(expectedBlogOwnerCPK).and(PostCPK.RATING.gt(4)
+                        .and(PostCPK.TITLE.eq("Placeholder title")))));
+        assertTrue(posts.contains(post));
+    }
+
+
+    /**
+     * Test that querying the saved item with multiple paths for foreign keys to the same
+     * model with missing data for intermediate model in join path and Custom Primary Key
+     * correctly forms the where clause and returns expected data.
+     * @throws DataStoreException On unexpected failure manipulating items in/out of DataStore
+     */
+    @Test
+    public void queryMultipleJoinWithMultipleRecordsWithMultipleConditionsAndCPK()
```

Comment:
Can you adjust this query to use the composite key: https://docs.amplify.aws/android/build-a-backend/more-features/datastore/customize-primary-keys/#querying-records-with-custom-primary-keys